File: app/src/main/java/com/tomaschlapek/tcbasearchitecture/widget/DataBindingAdapters.java
package com.tomaschlapek.tcbasearchitecture.widget;

import android.databinding.BindingAdapter;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.widget.ImageView;

/**
 * Created by tomaschlapek on 17/2/17.
 */

public class DataBindingAdapters {

  @BindingAdapter("android:src")
  public static void setImageUri(ImageView view, String imageUri) {
    if (imageUri == null) {
      view.setImageURI(null);
    } else {
      view.setImageURI(Uri.parse(imageUri));
    }
  }

  @BindingAdapter("android:src")
  public static void setImageUri(ImageView view, Uri imageUri) {
    view.setImageURI(imageUri);
  }

  @BindingAdapter("android:src")
  public static void setImageDrawable(ImageView view, Drawable drawable) {
    view.setImageDrawable(drawable);
  }

  @BindingAdapter("android:src")
  public static void setImageResource(ImageView imageView, int resource) {
    imageView.setImageResource(resource);
  }
}

File: app/src/main/kotlin/com/tomaschlapek/tcbasearchitecture/util/DatabindingAdapters.kt
package com.tomaschlapek.tcbasearchitecture.util

import android.databinding.BindingAdapter
import android.graphics.drawable.Drawable
import android.net.Uri
import android.widget.ImageView

/**
 * Adapters for Google data binding.
 */

@BindingAdapter("android:src")
fun setImageUri(view: ImageView, imageUri: String?) {
  if (imageUri == null) {
    view.setImageURI(null)
  } else {
    view.setImageURI(Uri.parse(imageUri))
  }
}

@BindingAdapter("android:src")
fun setImageUri(view: ImageView, imageUri: Uri) {
  view.setImageURI(imageUri)
}

@BindingAdapter("android:src")
fun setImageDrawable(view: ImageView, drawable: Drawable) {
  view.setImageDrawable(drawable)
}

@BindingAdapter("android:src")
fun setImageResource(imageView: ImageView, resource: Int) {
  imageView.setImageResource(resource)
}